### PR TITLE
[ROX-18761] : Remove report metadata table

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

This PR removes the report metadata proto and table. It replaces all of its usages with report snapshot. It also does some refactoring.

Renaming report_snapshots table to report_history coming in a followup PR.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Unit testing

Manual testing

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/reports/scheduler/v2/reportgenerator/report_gen_impl.go [hunk 1]
```diff
@@ -373,17 +368,17 @@ func (rg *reportGeneratorImpl) getWatchedImages(ctx context.Context) ([]string,
 	return results, nil
 }
 
-func (rg *reportGeneratorImpl) upsertReportStatus(ctx context.Context, metadata *storage.ReportMetadata, status storage.ReportStatus_RunState) error {
-	metadata.ReportStatus.RunState = status
-	return rg.reportMetadataStore.UpdateReportMetadata(ctx, metadata)
+func (rg *reportGeneratorImpl) upsertReportStatus(ctx context.Context, snapshot *storage.ReportSnapshot, status storage.ReportStatus_RunState) error {
```
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/report_gen_impl.go:371]: That is correct. `updateStatus` would be a better name here.

File: central/reports/scheduler/v2/reportgenerator/report_gen_impl.go [hunk 2]
```diff
@@ -392,57 +387,13 @@ func (rg *reportGeneratorImpl) logAndUpsertError(reportErr error, req *ReportReq
 		return
 	}
 	log.Errorf("Error while running report for config '%s': %s", req.ReportConfig.GetName(), reportErr)
-	req.ReportMetadata.ReportStatus.ErrorMsg = reportErr.Error()
-	req.ReportMetadata.ReportStatus.CompletedAt = types.TimestampNow()
-	err := rg.upsertReportStatus(req.Ctx, req.ReportMetadata, storage.ReportStatus_FAILURE)
+	req.ReportSnapshot.ReportStatus.ErrorMsg = reportErr.Error()
```
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/report_gen_impl.go:390]: Doing the nil check on `req.ReportSnapshot.ReportStatus` above this line.

File: central/reports/scheduler/v2/reportgenerator/validate.go [hunk 3]
```diff
@@ -19,12 +19,20 @@ func ValidateReportRequest(request *ReportRequest) error {
 		return errors.New("Report configuration ID is empty")
 	}
 
-	if request.ReportMetadata == nil || request.ReportMetadata.ReportStatus == nil {
-		return errors.New("Report request does not have a valid report metadata with report status")
+	if request.Collection == nil {
```
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/validate.go:22]: The `request == nil ` check happens at the top of this function. If request is nil, it short circuits and returns without doing other checks.

```
func ValidateReportRequest(request *ReportRequest) error {
	if request == nil {
		return errors.New("Report request is nil.")
	}
	
	if request.ReportConfig == nil {
		return errors.New("Report request does not have a valid non-nil report configuration")
	}
	...
	...
}
```

**Review by md2119 — APPROVED** [review]
Please address the comments before merging. I am happy to see this simplification of the framework.

File: central/reports/scheduler/v2/reportgenerator/validate.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/validate.go:23]: Ah okay I see what you mean.

File: central/reports/scheduler/v2/reportgenerator/report_gen_impl.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/reports/scheduler/v2/reportgenerator/report_gen_impl.go:391]: Got it. Done.

**Pull request merged**